src/azure-firewall/azext_firewall/custom.py

```diff
@@ -78,26 +90,43 @@ def create_azure_firewall(cmd, resource_group_name, azure_firewall_name, locatio
                              virtual_hub=SubResource(id=virtual_hub) if virtual_hub is not None else None,
                              firewall_policy=SubResource(id=firewall_policy) if firewall_policy is not None else None,
                              sku=sku_instance if sku is not None else None,
-                             threat_intel_mode=threat_intel_mode)
+                             threat_intel_mode=threat_intel_mode,
+                             hub_ip_addresses=HubIPAddresses(
+                                 public_ips=HubPublicIPAddresses(
+                                     count=hub_public_ip_count
+                                 )
+                             ) if hub_public_ip_count is not None else None)
     if private_ranges is not None:
         if firewall.additional_properties is None:
             firewall.additional_properties = {}
         firewall.additional_properties['Network.SNAT.PrivateRanges'] = private_ranges
-
-    firewall.additional_properties['Network.DNS.EnableProxy'] = \
-        enable_dns_proxy if enable_dns_proxy is not None else False
-    firewall.additional_properties['Network.DNS.RequireProxyForNetworkRules'] = \
-        require_dns_proxy_for_network_rules if require_dns_proxy_for_network_rules is not None else True
-    firewall.additional_properties['Network.DNS.Servers'] = ','.join(dns_servers or '')
+    if sku is None or sku.lower() == 'azfw_vnet':
+        if firewall_policy is None:
+            firewall.additional_properties['Network.DNS.EnableProxy'] = \
+                enable_dns_proxy if enable_dns_proxy is not None else False
+            firewall.additional_properties['Network.DNS.RequireProxyForNetworkRules'] = \
+                require_dns_proxy_for_network_rules if require_dns_proxy_for_network_rules is not None else True
```

Comment:
default is true?

Reply:
I think so. It's done by @haroldrandom. I invited you into a group chat for the context.